### DISCUSSION TITLE
No color setting possible inside the conky window

### DISCUSSION THREAD

**Discussion by Stefan2506** [brndnmtthws/conky#1854] [Q&A] [answered]
Dears,

might be this is a really small issue, I may just overlook something, but after about two weeks of trying, I have no further idea.
If I behave stupidly, feel free to beat me :-)

I am running conky on a Raspberry PI 4B with an installed X11 and Lightdm I have disabled Wayland so I am really running X
This is the OS installed: Linux squeeze 6.6.20+rpt-rpi-v8 #1 SMP PREEMPT Debian 1:6.6.20-1+rpt1 (2024-03-07) aarch64 GNU/Linux
Debian Bookworm
The PI has a 3.5" Touchscreen, X comes up with all colors, so background window, cursor, the raspberry as a start button is really red... and so on.
So color support is there.
Display is able to show the normal color range display usually show.

Conky is running fine, apart from I am unable to change the color settings inside the conky window.
For the window itself, I can set any color I like and it works.
I tried pink, white, grey... others more, all fine.
But if I set a color in the text area, it will be ignored, whatever I tried.
I tried:
Setting the colors with variables like below.
-- Colors
    default_color = '#D3D3D3'                   -- Light Grey
    default_outline_color = '#000000'           -- Black
    default_shade_color = '#000000'             -- Black
    color1 = 'FF8C00'                           -- Dark Orange
    color2 = 'DC143C'                           -- Crimoson Red
    color3 = 'FF4500'                           -- Orange Red
    color4 = 'FFA500'                           -- Orange
    color5 = '00008B'                           -- Dark Blue 

conky.text = [[
${color1}Conky ${Conky_version} $sysname $nodename $kernel $machine
${color2}Frequency (in Ghz):$freq_g ${color yellow}Uptime:$color $uptime
${color3}CPU Usage:  $cpu% ${cpubar 4}
${color4}RAM Usage:  $mem/$memmax - $memperc% ${membar 4}
${color5}Swap Usage: $swap/$swapmax - $swapperc% ${swapbar 4}
]]

All text... etc. will be shown in black.
If I set the window background to black, everything disappears.
I tried hex with the '#' and without.
I tried names accourding to /usr/X11/rgb.txt.
Result is always the same, it'll come up with black text and that's it.

I also tried directly in the text like shown below (different tries so different examples)
conky.text = [[
${"DarkOrange"}Conky ${Conky_version} $sysname $nodename $kernel $machine
${ '#DC143C'}Frequency (in Ghz):$freq_g ${color yellow}Uptime:$color $uptime
${"Orange"}CPU Usage:  $cpu% ${cpubar 4}
${"Dark Blue"}RAM Usage:  $mem/$memmax - $memperc% ${membar 4}
${'FFA500'}Swap Usage: $swap/$swapmax - $swapperc% ${swapbar 4}
]]

Result: See above.
All text and bars are shown in black, no change whatever I tried.

Apart from the Windows color it seems no other color is set.
I do not get an error message when conky starts, I only have this one /etc/conky/conky.conf.

I have tried some about 10 different configs from pages where one can download conky configs...
nothing. No color in the window.

Please find my current config below

conky.config = {

-- Various settings
    background = true,
    double_buffer = true,
    cpu_avg_samples = 2,
    net_avg_samples =2,
    net_avg_samples = 2,
    format_human_readable = true,
    no_buffers = true,
    out_to_console = false,
    out_to_ncurses = false,
    out_to_stderr = false,
    out_to_x = true,
    out_to_wayland = false,
    total_run_times = 0,
    update_interval = 5.0,

-- Placement
    alignment = 'top_middle',
    gap_x = 0,
    gap_y = 0,
    maximum_width = 640,
    minimum_height = 480,
    minimum_width = 630,

-- Graphical
    border_width = 1,
    draw_borders = false,
    draw_graph_borders = false,
    draw_outline = false,
    draw_shades = false,
    show_graph_range = false,
    show_graph_scale = false,
    stippled_borders = 0,

-- Window
    own_window = true,
    own_window_class = 'Conky',
    own_window_type = 'desktop',
    own_window_transparent = false,
    own_window_colour = 'FFF5EE',

-- Textual
    extra_newline = false,
    font = 'DejaVu Sans Mono:size=12',
    uppercase = false,
    use_spacer = 'none',
    use_xft = true,
}

This is the conky versions and what was compiled in:
conky 1.18.3 compiled 2023-03-07 for Linux armv7l
Compiled in features:
System config file: /etc/conky/conky.conf
Package library path: /usr/lib/conky
 General:
  * math
  * hddtemp
  * portmon
  * IPv6
  * wireless
  * support for IBM/Lenovo notebooks
  * builtin default configuration
  * old configuration syntax
  * Imlib2
  * OSS mixer support
  * apcupsd
  * iostats
  * ncurses
  * Internationalization support
 X11:
  * Xdamage extension
  * Xinerama extension (virtual display)
  * Xshape extension (click through)
  * XDBE (double buffer extension)
  * Xft
  * ARGB visual
  * Own window
  * Mouse evenets
 Music detection:
  * CMUS
  * MPD
  * MOC
 Default values:
  * Netdevice: eno1
  * Local configfile: $HOME/.conkyrc
  * Localedir: /usr/share/locale
  * Maximum netdevices: 256
  * Maximum text size: 16384
  * Size text buffer: 256

Any hint?
Anything you would need to know?

Many thanks in advance for your kind help.
Stefan

**Comment by rschell** [comment, marked as answer]
This sounds similar to #1477, which was fixed in #1768.  

That change would appear in versions after 1.20.0.  That version hasn't made it into "sid" or "testing" yet, so to confirm you would have to build conky from the git repo.

**Comment by Caellian** [comment]
> If I behave stupidly, feel free to beat me :-)

We're not allowed to under the contributing by-laws. :smile:
No, don't fret to ask questions, I don't know full conky functionality even though I went through 60% of its code (and understand 40%).

> so to confirm you would have to build conky from the git repo.

You could also download and unpack the [AppImage](https://github.com/brndnmtthws/conky/releases/tag/v1.20.1), but it might not work (depending on system library versions). Compiling is straightforward though, see [wiki](https://github.com/brndnmtthws/conky/wiki/Installation).

**Comment by rschell** [comment]
Using 1.20.2_pre, I get:

![image](https://github.com/brndnmtthws/conky/assets/1888300/5eac08ff-2ca2-42bc-87f3-f4dc3fdc6ae9)

here is my version of the .conkyrc file:
[.conkyrc.txt](https://github.com/brndnmtthws/conky/files/15066498/default.conkyrc.txt)

**Comment by Stefan2506** [comment]
Thanks all for the swift response.
I will try to compile it myself and then report about my conky experience.
May take some days, but I will.

Stefan
